src/easylink/step.py

```diff
@@ -1300,63 +1325,23 @@ def set_configuration_state(
             The configuration for any implementations to be combined.
         input_data_config
             The input data configuration for the entire pipeline.
-
-        Notes
-        -----
-        We update the :class:`easylink.graph_components.StepGraph` and ``SlotMappings``
-        in :meth:`validate_step` (as opposed to in :meth:`set_configuration_state`
-        as is done in :class:`TemplatedStep`) because :meth:`validate_step` is called
-        prior to :meth:`set_configuration_state`, but the validations itself actually
-        requires the updated ``StepGraph`` and ``SlotMappings``.
         """
+        chosen_config = self.choices[step_config["type"]]
+        self.step_graph = StepGraph()
+        self.step_graph.add_node_from_step(chosen_config["step"])
+        self.slot_mappings = {
+            "input": chosen_config["input_slot_mappings"],
+            "output": chosen_config["output_slot_mappings"],
+        }
```

Comment:
I will note, I'm a bit surprised that `slot_mappings` remains as a property on base `Step`. I think I haven't fully wrapped my mind around what they are doing for us in the leaf case.

Reply:
I think it's just that it's used by both HierarchicalSteps and TemplatedSteps?

Reply:
Mappings seem to be used in LeafConfigurationState, not only NonLeaf: https://github.com/ihmeuw/easylink/blob/bae3518e6638502be6dc29832cdf2fae109d6280/src/easylink/step.py#L1495

Reply:
Ah, right, we use slot mappings to (1) remap step-level slots (like from the main input of a single HierarchicalStep to the main input of the first of its sub-steps) as well as (2) mapping _Implementation_ nodes (which is why I think it needs to stick around in `Step`.

@patricktnast ?

Reply:
Hmm interesting, maybe this confusion comes down to a clarification needed in the `SlotMapping` docstring, as that only seems to cover (1):

https://github.com/ihmeuw/easylink/blob/b24f85ae7b3c6778199857148b53403d283177b6/src/easylink/graph_components.py#L259-L264

Reply:
I did at the note "Nodes can be either :class:`Steps<easylink.step.Step>` or :class:`Implementations<easylink.implementation.Implementation>`." Is that not clear? Or is it but you'd prefer it to not be in the Notes section?

Reply:
Ok, I tweaked the docstring a bit